### ISSUE TITLE
Refresh explorer button changes sort order

### ISSUE THREAD

**Issue by dianneman** [Microsoft/vscode#72474]
I've set "explorer.sortOrder": "modified" but the explorer does switch back to default sort order. 

- VSCode Version:1.33.1
- OS Version:Windows 10 1809

Steps to Reproduce:

1.When I push multiple times the "Refresh explorer" button in explorer on an opened folder the sort order changes between default and modified randomly. It can reproduced by fast clicking the button, the last order will persist.

Does this issue occur when all extensions are disabled?: Yes

**Comment by isidorn** [comment]
I can not reproduce this.
Can you reproduce this with vscode insiders
https://code.visualstudio.com/insiders/

**Comment by dianneman** [comment]
I can also reproduce it in this insider build. Installed insiders, went to open settings and only have one setting set, the one I mentioned. It seems a fast double click does the trick after a few times. Probably you need more items inside your test folder? My folder (open folder) consist of 15 folders and 195 .ps files in open folder.

**Comment by isidorn** [comment]
I can reproduce now. But only on windows slow machine.
Currently I do not have cycles to look into this. A PR that fixes this is welcome.
Code pointer https://github.com/Microsoft/vscode/blob/585f85399738cf188d356d5cbb0b8e24e1e7e6d0/src/vs/workbench/contrib/files/common/explorerService.ts#L41

Whoever wants to fix this should first
1) Be able to reproduce this consistnently
2) Check out code pointer and the issue is probably timing related how the sortOrder attribute is being read when resolving stats

I will not accept some complex fixes. So the PR should be straight forward.
Thanks

**Comment by dianneman** [comment]
Good to read you can reproduce. I'm using a Dell Xps 15 -i7-32 GB, faster then Elon Musks Falcon rocket :D

**Comment by jmbockhorst** [comment]
I will look into this.

**Comment by jmbockhorst** [comment]
I found that the sort order would reset every time I opened the settings page again. 

I would choose modified time as the sort order, refresh the exporer and close the settings. It would be sorted correctly. Then when I would open the settings again, the sort order would reset to the default order, until I refresh the explorer again. This may be a different problem that what you were describing.

![sortOrder](https://user-images.githubusercontent.com/44308390/56299044-def0f500-60f8-11e9-98b4-3b6d9a02ae49.gif)
